src/neural/network_check.cc

```diff
@@ -77,48 +99,114 @@ class CheckComputation : public NetworkComputation {
 
     bool policyAlmostEqual = true;
     for (int i = 0; i < size && policyAlmostEqual; i++) {
-      for (int j = 0; j < NUM_OUTPUT_POLICY; j++) {
+      for (int j = 0; j < kNumOutputPolicies; j++) {
         float v1 = refComp_->GetPVal(i, j);
         float v2 = checkComp_->GetPVal(i, j);
         policyAlmostEqual &= IsAlmostEqual(v1, v2);
       }
     }
 
-    // We could also print on standard output using the uci info string
-    // example:
-    //
-    // info string Check passed
-    // info string ***** error check failed
-    //
-
     if (valueAlmostEqual && policyAlmostEqual) {
-      std::cerr << "Check passed for a batch of " << size << std::endl;
+      fprintf(stderr, "Check passed for a batch of %d.\n", size);
       return;
     }
 
     if (!valueAlmostEqual && !policyAlmostEqual) {
-      std::cerr << "*** ERROR check failed for a batch of " << size
-                << " both value and policy incorrect" << std::endl;
+      fprintf(stderr,
+              "*** ERROR check failed for a batch of %d  both value and policy "
+              "incorrect.\n",
+              size);
       return;
     }
 
     if (!valueAlmostEqual) {
-      std::cerr << "*** ERROR check failed for a batch of " << size
-                << " value incorrect (but policy ok)" << std::endl;
+      fprintf(stderr,
+              "*** ERROR check failed for a batch of %d value incorrect (but "
+              "policy ok).\n",
+              size);
       return;
     }
 
-    std::cerr << "*** ERROR check failed for a batch of " << size
-              << " policy incorrect (but value ok)" << std::endl;
+    fprintf(stderr,
+            "*** ERROR check failed for a batch of %d policy incorrect (but "
+            "value ok).",
+            size);
+  }
+
+  bool IsAlmostEqual(double a, double b) {
```

Comment:
const?

Also, why double? float to double conversion is slow, so there should be a strong reason to use double as the rest of the code uses float.
It also causes warnings (at least in MSVS), which will soon be treated as errors.

Reply:
Is that really important on private members? Yet addressed.